packages/optimizer/lib/transformers/PreloadHeroImage.js

```diff
@@ -16,331 +16,7 @@
-        isValidImageSrcURL(child.attribs.src)
-      ) {
-        return {
-          ampImg: child,
-          src: child.attribs.src,
-          media: ampIframe.attribs.media,
-          srcset: child.attribs.srcset || '',
-        };
-      }
-    }
-    return null;
-  }
-
-  // Checks if node qualifies to be a hero image. Returns HeroImageSrcs if the
-  // node is a hero image. The hero image here can come from one of <amp-img>,
-  // <amp-video>, <amp-iframe>, <amp-video-iframe>.
-  isCandidateImageForPreloading(ampImg) {
-    const src = ampImg.attribs.src;
-    if (!src) {
-      return null;
-    }
-    if (!isValidImageSrcURL(src)) {
-      return null;
-    }
-
-    let {width, height, srcset, layout, media} = ampImg.attribs;
-
-    if (!width && !height) {
-      if (layout === 'fill') {
-        ({width, height} = this.nodeDimensionsFromParent(ampImg));
-      } else {
-        return null;
-      }
-    }
-    if (this.isTinyNode(layout, width, height)) {
-      return null;
-    }
-    return {ampImg, src, srcset, media};
-  }
-
-  // Any node with width or height less than 150 pixels and a non-responsive layout.
-  isTinyNode(layout, width, height) {
-    if (width <= 0 || height <= 0) return true;
-    if (layout === 'intrinsic' || layout === 'responsive') {
-      return false;
-    }
-    return width < TINY_IMG_THRESHOLD || height < TINY_IMG_THRESHOLD;
-  }
-
-  nodeDimensionsFromParent(node) {
-    while (node.parent) {
-      node = node.parent;
-      if (!node.attribs) {
-        continue;
-      }
-      const width = node.attribs.width;
-      const height = node.attribs.height;
-      if (!width && !height) {
-        continue;
-      }
-      return {width, height};
-    }
-    return {width: 0, height: 0};
-  }
-
-  generateImg(node) {
-    if (!node) {
-      return;
-    }
-    node.attribs['i-amphtml-ssr'] = '';
-    node.attribs['data-hero'] = '';
-    // Create img node
-    const imgNode = createElement('img', {
-      class: 'i-amphtml-fill-content i-amphtml-replaced-content',
-      decoding: 'async',
-    });
-    // Copy attributes
-    const attributesToCopy = [
-      'alt',
-      'attribution',
-      'referrerpolicy',
-      'src',
-      'srcset',
-      'sizes',
-      'title',
-    ];
-    for (const attr of attributesToCopy) {
-      if (hasAttribute(node, attr)) {
-        imgNode.attribs[attr] = node.attribs[attr];
-      }
-    }
-    // Generate styles for object-fit and object-position
-    const styles = [];
-    const objectFit = node.attribs['object-fit'];
-    if (objectFit) {
-      styles.push(`object-fit:${objectFit}`);
-    }
-    const objectPosition = node.attribs['object-position'];
-    if (objectPosition) {
-      styles.push(`object-position:${objectPosition}`);
-    }
-    if (styles.length > 0) {
-      imgNode.attribs.style = styles.join(';');
-    }
-    appendChild(node, imgNode);
-  }
-}
+/* This transfomer has been deprecated, use OptimizeHeroImages.js instead */
```

Comment:
who is this message for? Should it be logged, rather than a comment?

Reply:
Added logging for the options. We don't have access to the log here.